agent/connect/csr.go

```diff
@@ -47,11 +48,28 @@ func SigAlgoForKeyType(keyType string) x509.SignatureAlgorithm {
 // along with the PEM-encoded private key for this certificate.
 func CreateCSR(uri CertURI, privateKey crypto.Signer,
 	dnsNames []string, ipAddresses []net.IP, extensions ...pkix.Extension) (string, error) {
+
+	// Drop everything after the ':' from the name when constructing the DNS SANs.
+	uniqueNames := make(map[string]struct{})
+	formattedDNSNames := make([]string, 0)
+	for _, host := range dnsNames {
+		hostSegments := strings.Split(host, ":")
+		formattedHost := hostSegments[0]
```

Comment:
Should we confirm that `len(hostSegments) > 0` here before accessing the 0th element, to ensure we don't have an out-of-bounds access?

Reply:
To test this edge case, I guess you could add a host `""` to the array in your test

Reply:
This should be okay. [`strings.Split`](https://pkg.go.dev/strings#Split) only returns an empty list when both arguments are empty.

`formattedHost` could be empty here, but that would only occur if the one of `dnsNames` had no hostname, like `:8080`.

Reply:
We wouldn't want a bad configuration (like "`:8080`") to cause a panic though, right?